test/jasmine/tests/legend_scroll_test.js

```diff
@@ -324,30 +323,32 @@ describe('The legend', function() {
             scrollBarX = scrollBar.getAttribute('x'),
             scrollBarY = scrollBar.getAttribute('y');
 
-            Plotly.relayout(gd, 'showlegend', false);
-            Plotly.relayout(gd, 'showlegend', true);
-
-            legend = getLegend();
-            scrollBox = getScrollBox();
-            scrollBar = getScrollBar();
-            toggle = getToggle();
-
-            legend.dispatchEvent(scrollTo(wheelDeltaY));
-            expect(scrollBar.getAttribute('x')).toBe(scrollBarX);
-            expect(scrollBar.getAttribute('y')).toBe(scrollBarY);
-
-            var dataScroll = getScroll(gd);
-            toggle.dispatchEvent(new MouseEvent('mousedown'));
-            toggle.dispatchEvent(new MouseEvent('mouseup'));
-            setTimeout(function() {
-                expect(+toggle.parentNode.style.opacity).toBeLessThan(1);
-                expect(getScroll(gd)).toBe(dataScroll);
-                expect(scrollBox.getAttribute('transform')).toBe(
-                    'translate(0,' + -dataScroll + ')');
-                expect(scrollBar.getAttribute('width')).toBeGreaterThan(0);
-                expect(scrollBar.getAttribute('height')).toBeGreaterThan(0);
-                done();
-            }, DBLCLICKDELAY * 2);
+            Plotly.relayout(gd, 'showlegend', false)
+            .then(function() {
+                Plotly.relayout(gd, 'showlegend', true);
+
+                legend = getLegend();
+                scrollBox = getScrollBox();
+                scrollBar = getScrollBar();
+                toggle = getToggle();
+
+                legend.dispatchEvent(scrollTo(wheelDeltaY));
+                expect(scrollBar.getAttribute('x')).toBe(scrollBarX);
+                expect(scrollBar.getAttribute('y')).toBe(scrollBarY);
+
+                var dataScroll = getScroll(gd);
+                toggle.dispatchEvent(new MouseEvent('mousedown'));
+                toggle.dispatchEvent(new MouseEvent('mouseup'));
+                setTimeout(function() {
+                    expect(+toggle.parentNode.style.opacity).toBeLessThan(1);
+                    expect(getScroll(gd)).toBe(dataScroll);
+                    expect(scrollBox.getAttribute('transform')).toBe(
+                        'translate(0,' + -dataScroll + ')');
+                    expect(scrollBar.getAttribute('width')).toBeGreaterThan(0);
+                    expect(scrollBar.getAttribute('height')).toBeGreaterThan(0);
+                    done();
```

Comment:
I guess this wants the `new Promise` / `resolve` pattern you used a few other places. Also there's a second `Plotly.relayout` above that should get its own async step.

Reply:
Attempt addressing in https://github.com/plotly/plotly.js/pull/5340/commits/3a313e752bfea723f4370de857cc3c1bfdf6eb4b.

Reply:
Hmm.. after that change the tests started failing. So I revert the changes to this test in https://github.com/plotly/plotly.js/pull/5340/commits/5a127f0f3769a7aa006ae73028059450df0e2403 but we may come back to improve those in a separate PR.